src/herder/HerderImpl.cpp

```diff
@@ -314,12 +318,41 @@ HerderImpl::processExternalized(uint64 slotIndex, StellarValue const& value)
         }
     }
 
-    // tell the LedgerManager that this value got externalized
-    // LedgerManager will perform the proper action based on its internal
-    // state: apply, trigger catchup, etc
-    LedgerCloseData ledgerData(static_cast<uint32_t>(slotIndex),
-                               externalizedSet, value);
-    mLedgerManager.valueExternalized(ledgerData);
+    auto publishToArchive = [this, value, slotIndex]() {
```

Comment:
This change is too broad and probably not doing what you want:
* this delays externalizing ledgers in addition to delaying publish
* by not applying ledgers synchronously with `processExternalized`, I *think* you're potentially introducing new classes of race conditions where the node will potentially externalize the next slot before having applied the previous one. This in turn will cause all sorts of problems (like skipping applying ledgers as you only have one timer to apply ledgers)

Reply:
I see. My understanding now is that we should only delay publishing not externalising ledgers?

For that purpose, does it suffice to delay executing [these steps](https://github.com/stellar/stellar-core/blob/ee13db4b76b8d68cf534b4c4afb74a100ac9c25a/src/ledger/LedgerManagerImpl.cpp#L923C1-L929C1)?